extensions/amp-social-share/1.0/social-share.type.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @externs */
+
+/**
+ * @typedef {{
+ *   type:       ?string,
+ *   endpoint:   ?string,
```

Comment:
I think all of these are actually `type|undefined` and not `?type`. In other words - this prop can be omitted, but when specified, the valid value should be given. And `null` typically is not valid.

Reply:
Thanks, updated in the latest revision.  I had confused `nullable` and `|undefined`.